magenta-lib/src/main/scala/magenta/tasks/AWS.scala

```diff
@@ -240,14 +241,24 @@ object ASG {
     autoScalingGroups.headOption.getOrElse(reporter.fail(s"Failed to identify an autoscaling group with name ${name}"))
   }
 
-  def groupForAppAndStage(pkg: DeploymentPackage, stage: Stage, stack: Stack, client: AutoScalingClient, reporter: DeployReporter): AutoScalingGroup = {
+  def groupForAppAndStage(pkg: DeploymentPackage, stage: Stage, stack: Stack, cdkTagRequirements: Option[CdkTagRequirements], client: AutoScalingClient, reporter: DeployReporter): AutoScalingGroup = {
     case class ASGMatch(app:App, matches:List[AutoScalingGroup])
 
     implicit class RichAutoscalingGroup(asg: AutoScalingGroup) {
       def hasTag(key: String, value: String): Boolean = asg.tags.asScala.exists { tag =>
         tag.key == key && tag.value == value
       }
-      def matchApp(app: App, stack: Stack): Boolean = hasTag("Stack", stack.name) && hasTag("App", app.name)
+      def meetsCdkTagRequirements(cdkTagRequirements: Option[CdkTagRequirements]) = {
+        val cdkTagIsPresent: Boolean = asg.tags.asScala.exists { tag => tag.key == "gu:cdk:pattern-name" }
```

Comment:
For extra clarity in the future, might be worth adding some comments around this? Or linking to the PR that added the tag?